fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -27,23 +27,87 @@
  * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
  */
 public class FabricKeyBinding extends KeyBinding {
+
+    private final Identifier id;
+
 	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
 		super("key." + id.toString().replace(':', '.'), type, code, category);
+		this.id = id;
+	}
+
+	/**
+	 * Original identifier used to register this key.
+	 * May be different from the {@link getId()}!
+	 */
+	public Identifier getIdentifier() {
+	    return id;
 	}
 
 	public static class Builder {
-		protected final FabricKeyBinding binding;
 
-		protected Builder(FabricKeyBinding binding) {
-			this.binding = binding;
+		private InputUtil.Type type = InputUtil.Type.KEYSYM;
+
+		private Identifier id = new Identifier("fabric", "unnamed");
+
+		private int code;
+
+		private String category = KeyCategory.MISC;
+
+		private Builder() {
+
+		}
+
+		public Builder id(Identifier keyName) {
+		    this.id = keyName;
+			return this;
+		}
+
+		public Builder category(String category) {
+			this.category = category;
+			return this;
+		}
+
+		public Builder code(int keyCode) {
+			this.code = keyCode;
+			return this;
+		}
+
+		public Builder type(InputUtil.Type type) {
+			this.type = type;
+			return this;
 		}
 
 		public FabricKeyBinding build() {
-			return binding;
+			return new FabricKeyBinding(id, type, code, category);
+		}
+
+		public static Builder create() {
+			return new Builder();
 		}
 
+		/**
+		 * @deprecated Prefer using the builder itself.
+		 * <pre><code>
+		 * FabricKeyBinding.Builder builder = new FabricKeyBinding.Builder();
+		 * FabricKeyBinding left = builder
+		 *                            .id(new identifier("example", "left"))
+	     *                            .code(Keys.Left)
+	     *                            .build();
+         * FabricKeyBinding right = builder
+         *                            .id(new identifier("example", "right"))
+         *                            .code(Keys.Right)
+         *                            .build();
+         * KeyBindingRegistry.register(left);
+         * KeyBindingRegistry.register(right);
+         * </code></pre>
+		 */
+		@Deprecated
 		public static Builder create(Identifier id, InputUtil.Type type, int code, String category) {
-			return new Builder(new FabricKeyBinding(id, type, code, category));
+			return new Builder()
+					.id(id)
+					.type(type)
+					.code(code)
+					.category(category);
```

Comment:
This chain is short enough to be declared in a single line

Reply:
I'll defer this one to @killjoy1121. All I know is he tells me they have to be on separate lines otherwise kittens and puppies get hurt.

Reply:
You think I'm the one in charge? You're the one who brought me here.